types/express-session/index.d.ts

```diff
@@ -339,10 +339,10 @@ declare namespace session {
 
         /** Returns all sessions in the store */
         // https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38783, https://github.com/expressjs/session/pull/700#issuecomment-540855551
-        all?(callback: (err: any, obj?: SessionData[] | { [sid: string]: SessionData; } | null) => void): void;
+        all?(callback: (err: any, obj?: SessionData[] | { [sid: string]: SessionData } | null) => void): void;
 
         /** Returns the amount of sessions in the store. */
-        length?(callback: (err: any, length: number) => void): void;
+        length?(callback: (err: any, length?: number) => void): void;
```

Comment:
isn't the underlying store always returning a number? e.g.:
```ts
callback(null, Object.keys(sessions).length)
```
why is this change introduced?

Reply:
The implementations within the `express-session` repo show that these callbacks are called without `length` being provided when an error is present. It's a consistent pattern within the repo to drop the second argument in an error condition. I mentioned it in the commit message, but probably could have been more explicit about it. Here's the related link: https://github.com/expressjs/session/blob/master/session/memory.js#L133

Reply:
that's ok and formally correct. Though in TS that's not recommended (to be very terse here with syntax), as the nature of JS (TS per extension) is to always allow optional params in callbacks:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#optional-parameters-in-callbacks

Reply:
That's good to know. It feels like an area of ambiguity that could be problematic in practice. But maybe I need to do more reading on TypeScript and specific cases like this. Thank you for the link!

What's the appropriate course to take here? I can remove the optionality change for the `length` parameter. A caveat is that the other callbacks here have the same optionality, so it would remain inconsistent. Should those change at some point?

Reply:
@jneander the docs (DO/DON'T) contain guides, not the golden bullet solution. I think you're correct here, checked the other methods definition, they allow to ease strictness when defining callbacks (no matter what is the nature of TS, etc). The callbacks seem to be defined with optional params, with `length` as exception, which is weird (as not aligned with intent in other methods).
Looking good, thanks for the patience :)

Reply:
No worries! Thank you for the PR review. I'm happy to contribute, and want to make sure I'm following preferred patterns where expected. If there's anything else you need from me on this, please let me know.